backend/utils/debiai/widgetConfigurations.py

```diff
@@ -0,0 +1,119 @@
+import os
+import json
+import utils.utils as utils
+import uuid
+
+CONF_PATH = "data/widgetConfigurations.json"
+
+# Configuration file structure
+# {
+#     "widgetKey": [
+#         {
+#             "id": ""
+#             "name": ""
+#             "description": "",
+#             "projectId": ""
+#             "dataProviderId": ""
+#             "creationDate": 0
+#
+#             "configuration": {},
+#         },
+#         ...
+#      ]
+# }
+
+
+def setup_widget_configurations():
+    # Create the folder if it does not exist
+    if not os.path.exists("data"):
+        os.mkdir("data")
+
+    # Create the file if it does not exist
+    if not os.path.exists(CONF_PATH):
+        _save_configurations({})
+
+
+def get_configurations_overview():
+    # Return the number of configurations for each widget
+    all_configurations = _get_all_configurations()
+
+    configurations_overview = {}
+    for widget_key in all_configurations:
+        configurations_overview[widget_key] = len(all_configurations[widget_key])
+
+    return configurations_overview
+
+
+def get_configurations(widget_key):
+    # Return the configurations list of the widget
+    all_configurations = _get_all_configurations()
+
+    if widget_key in all_configurations:
+        return all_configurations[widget_key]
+    else:
+        return []
+
+
+def add_configuration(widget_key, data):
```

Comment:
I think we should check for the conf name and return error if the value contain special char